octavia-cli/octavia_cli/apply/resources.py

```diff
@@ -202,6 +205,14 @@ def __init__(
     def remote_resource(self):
         return self._get_remote_resource()
 
+    def _get_comparable_configuration(
+        self,
+    ):  # pragma: no cover
+        if not self.was_created:
+            raise NonExistingResourceError("Can't find a comparable configuration as the remote resource does not exists.")
+        else:
+            return copy.deepcopy(self.remote_resource)
+
     @property
     def was_created(self):
         return True if self.remote_resource else False
```

Comment:
I saw other places where the self.was_created is invoked and throw errors. What do you think doing this directly in the `was_created` function?

Reply:
Do you mean raising the error in `was_created`? I'd like to use `was_created` as a boolean to know if the resource was created or not, I don't want it to throw an error if the resource was not created but simply return `false`. But maybe you meant running `copy.deepcopy(self.remote_resource)` in `was_created`? 🤔

Reply:
No, only the error handling in the `was_created` because is also invoked in connection generation command too

Reply:
We could maybe implement something like `self.remote_resource.raise_if_not_created()` ?